### DISCUSSION TITLE
CSSBattle scoring backend updates

### DISCUSSION THREAD

**Discussion by chinchang** [cssbattle/forum#248] [Announcements]
Players, scoring backend has been updated to latest Chromium.

## Known breaking changes:
1. `--` is not a valid variable name anymore. Existing solutions using -- will fail to render in the latest version of browser on your side and also won't render on the backend now. We'll soon put an outdated tag also on such solutions (wherever visible as open solutions)

**Comment by verberg** [comment]
Do i understand it correctly? It is no longer possible to top your own results?

**Reply by Beowolve** [reply]
True, only if you can reduce enough for the extra chars needed now.

**Reply by chinchang** [reply]
Well, I wouldn't say that. Its just that old solutions don't work anymore if they contain `--`. Your next best solution needs to be without `--`.

**Reply by rafehimself** [reply]
@chinchang  well not really ... some old solutions that were 100% now are 99.9% without `--`

**Reply by SkyHolder** [reply]
How about reset all progress and start from the scratch? Would that be fair?

**Reply by rafehimself** [reply]
@SkyHolder do you mean whoever will be online at the time of reset and submit old scores with 2 new additional chars, gets all the glory?
not really 😉

**Reply by verberg** [reply]
@Beowolve That's exactly the problem: Sometimes I'm so happy to reduce  one or two or three chars (you see I'm modest) to get more scores and now - no chance! It's annoying!

**Reply by SkyHolder** [reply]
I believe that the new version of the rules should lead to a new version of the best solutions, hence the new leaderboard

**Reply by rafehimself** [reply]
there is no rules here if you haven't noticed that yet

**Reply by SelenIT** [reply]
Sometimes new updates open new opportunities for shorter solutions. E.g. one of the updates broke the 109-char solution for Target 3, but another update enabled the 108-char solution that did only 99.9% match previously.

**Comment by APerricone** [comment]
so [cssbattle previewer](https://codepen.io/alexzaworski/full/vwmNKr) is no more valid?

**Reply by SelenIT** [reply]
I assume it's valid, as it uses the actual renderer (more or less:)

**Reply by rafehimself** [reply]
I wouldn't be so sure about it, might be misleading at the moment
that gets 100%
![image](https://user-images.githubusercontent.com/58576132/164799834-4da57436-b9ec-4b3e-b30e-23f165ce5b0f.png)
while something that looks more accurate not 😞 
![image](https://user-images.githubusercontent.com/58576132/164800224-f5e41852-00ca-4679-ad23-9f355c9174f7.png)

**Comment by chinchang** [comment]
Backend has been updated to latest Chromium - v102.

**Comment by chinchang** [comment]
Backend has been updated to latest Chromium - v104
